main.go

```diff
@@ -114,35 +114,75 @@ func main() {
 		os.Exit(1)
 	}
 
+	// Get non-flag command-line args
+	patterns := flag.Args()
+
 	// convert -skip flags to -ignore equivalents
 	for _, s := range skipExtensionFlags {
 		ignorePatterns = append(ignorePatterns, fmt.Sprintf("**/*.%s", s))
 	}
+
+	// create logger to print updates to stdout
+	logger := log.Default()
+
+	// real main
+	err := Run(
+		ignorePatterns,
+		spdx,
+		*holder,
+		*license,
+		*licensef,
+		*year,
+		*verbose,
+		*checkonly,
+		patterns,
+		logger,
+	)
+
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+// Run executes addLicense with supplied variables
+func Run(
+	ignorePatterns stringSlice,
+	spdx spdxFlag,
+	holder string,
+	license string,
+	licensef string,
+	year string,
+	verbose bool,
+	checkonly bool,
```

Comment:
`checkonly` gets a bit weird here, since this function currently calls `os.Exit(1)` if there are files that need changes, which is definitely not consumable. Perhaps an alternative would be to return a boolean that represents if changes are needed or were made. From there, a consumer of this function could infer that if they pass `checkonly = true` and get back a true, it means files need to be modified. If `checkonly = false` and they get back a true, it means files _were_ modified.

I don't love it, but I do like it more than returning an error. Just thinking out loud, though 🙂